### ISSUE TITLE
Unexpected parameter ordering in Detection class during iteration

### ISSUE THREAD

**Issue by alexandercarruthers** [roboflow/supervision#69]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar bug report.

### Bug

In the Detections class I expected to unpack the values in this order 

```python
for xyxy, mask, class_id, confidence, tracker_id in detections:
  ...   
```

due to the documentation string listing as 
```python
xyxy: np.ndarray
mask: np.Optional[np.ndarray] = None
class_id: Optional[np.ndarray] = None
confidence: Optional[np.ndarray] = None
tracker_id: Optional[np.ndarray] = None
```

but this works instead

```python
for xyxy, mask, confidence, class_id, tracker_id in detections:
  ...   
```

This is due to the `confidence` and `class_id` being in a different order in the `__iter__` function

```python
yield (
    self.xyxy[i],
    self.mask[i] if self.mask is not None else None,
    self.confidence[i] if self.confidence is not None else None,
    self.class_id[i] if self.class_id is not None else None,
    self.tracker_id[i] if self.tracker_id is not None else None,
)
```

Is there some consistency here that could be improved?

### Environment

- Supervision: 0.5.2

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by SkalskiP** [comment]
Hi @alexandercarruthers 👋🏻! You are absolutely right. What would you say for changing the order of properties in the class, not tuple elements in `__iter__`? So:

```python
xyxy: np.ndarray
mask: np.Optional[np.ndarray] = None
confidence: Optional[np.ndarray] = None
class_id: Optional[np.ndarray] = None
tracker_id: Optional[np.ndarray] = None
```
?

**Comment by SkalskiP** [comment]
The issue is fixed. Your PR is merged and changes released starting with version `0.6.0`.

**Comment by storm12t48** [comment]
Hello everyone
a little off topic of your bug how to retrieve the value of xyxy corresponding object detecter:
f"Target {xyxy}"
            for xyxy, mask, confidence, class_id, tracker_id in detections
